x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/deployment/DeploymentManager.java

```diff
@@ -116,7 +124,7 @@ private void doStartDeployment(TrainedModelDeploymentTask task, ActionListener<T
                 // here, we are being called back on the searching thread, which MAY be a network thread
                 // `startAndLoad` creates named pipes, blocking the calling thread, better to execute that in our utility
                 // executor.
-                executorServiceForProcess.execute(() -> startAndLoad(processContext, modelLoadedListener));
+                executorServiceForDeployment.execute(() -> startAndLoad(processContext, modelLoadedListener));
```

Comment:
This is a minor bug I think introduced by the allocation service change. This should be for deployment (starting named pipes, etc.) I think...Then the process context starts running within the process executor.

Reply:
++ i've made the same change locally for #75268